### ISSUE TITLE
linux: fs_fstat test fails with gyp build

### ISSUE THREAD

**Issue by bnoordhuis** [libuv/libuv#43]
v1.x dfdcfc0:

```
$ ./gyp_uv
['/home/bnoordhuis/src/libuv/uv.gyp', '-I', '/home/bnoordhuis/src/libuv/common.gypi', '--depth=.', '-f', 'make', '-Goutput_dir=/home/bnoordhuis/src/libuv/out', '--generator-output', '/home/bnoordhuis/src/libuv/out', '-Dgcc_version=49', '-Dclang=0', '-Dhost_arch=x64', '-Dtarget_arch=x64', '-Duv_library=static_library', '-Dcomponent=static_library']

$ make -C out BUILDTYPE=Debug
# elided

$ out/Debug/run-tests fs_fstat fs_fstat
Assertion failed in ../test/test-fs.c on line 1081: s->st_atim.tv_nsec == t.st_atim.tv_nsec
Aborted (core dumped)

$ file out/Debug/run-tests 
out/Debug/run-tests: ELF 64-bit LSB executable, x86-64, version 1 (SYSV), dynamically linked (uses shared libs), for GNU/Linux 2.6.32, BuildID[sha1]=f55d2937f14fa096a1e2bce1d6986f1671ecd261, not stripped

$ uname -a
Linux localhost.localdomain 3.17.3-300.fc21.x86_64 #1 SMP Fri Nov 14 23:36:19 UTC 2014 x86_64 x86_64 x86_64 GNU/Linux
```

The autotools build works, the test only fails when building with gyp.  A quick investigation suggests that neither _BSD_SOURCE, _SVID_SOURCE or _XOPEN_SOURCE are defined, making src/unix/fs.c skip the tv_nsec fields.

**Comment by saghul** [comment]
@bnoordhuis is this one also fixed by https://github.com/libuv/libuv/commit/fd3cf20a37b95fe8a3cb4be045b5a1bad9e53cde ?

**Comment by bnoordhuis** [comment]
Yes, sorry.  I knew there was another issue somewhere but I couldn't find it in the rush of things.  Fixed by fd3cf20, closing.
